### ISSUE TITLE
`goshimmer/client` runs unwanted code on `init()` when imported

### ISSUE THREAD

**Issue by dessaya** [iotaledger/goshimmer#1366]
## Bug description

GoShimmer client cannot be imported from third-party code, because it imports stuff from the goshimmer node, which in turn runs unwanted stuff on `init()`.

Eg:

```
github.com/iotaledger/goshimmer/client
  imports github.com/iotaledger/goshimmer/plugins/faucet
    imports github.com/iotaledger/goshimmer/plugins/autopeering/local
      has an init() function
```

```go
func init() {
	flag.String(CfgBind, "0.0.0.0", "bind address for global services such as autopeering and gossip")
	flag.String(CfgExternal, "auto", "external IP address under which the node is reachable; or 'auto' to determine it automatically")
	flag.Int(CfgPort, 14626, "UDP port for incoming peering requests")
	flag.String(CfgSeed, "", "private key seed used to derive the node identity; optional base58 or base64 encoded 256-bit string. Prefix with 'base58:' or 'base64', respectively")
}
```

The problem in our use case (the Wasp node) is that this ends up polluting the `pflag` parameters (apart from including unneeded stuff on our binary).

## GoShimmer version

- GoShimmer version: `develop`

## Steps To reproduce the bug

This example program reproduces the issue:

```go
package main

import (
	_ "github.com/iotaledger/goshimmer/client"
	"github.com/spf13/cobra"
)

var rootCmd = &cobra.Command{
	Version: "0.1.1",
	Use:     "test",
	Short:   "this is a test to show that goshimmer/client runs unwanted code on init()",
	Run: func(cmd *cobra.Command, args []string) {
		cmd.Help()
	},
}

func main() {
	err := rootCmd.Execute()
	if err != nil {
		panic(err)
	}
}
```

## Expected behaviour

The above command should print no flags on stdout.

## Actual behaviour

`go run main.go` produces the following output:

```
$ go run test/main.go
this is a test to show that goshimmer/client runs unwanted code on init()

Usage:
  test [flags]

Flags:
      --autopeering.entryNodes strings                    list of trusted entry nodes for auto peering (default [2PV5487xMw5rasGBXXWeqSi4hLz7r19YBt8Y1TGAsQbj@ressims.iota.cafe:15626,5EDH4uY78EA6wrBkHHAVBWBMDt7EcksRq6pjzipoW15B@entryshimmer.tanglebay.com:14646])
      --autopeering.networkVersion int                    autopeering network version (default 31)
      --autopeering.port int                              UDP port for incoming peering requests (default 14626)
      --autopeering.seed string                           private key seed used to derive the node identity; optional base58 or base64 encoded 256-bit string. Prefix with 'base58:' or 'base64', respectively
  -c, --config string                                     file path of the config file (default "config.json")
      --database.directory string                         path to the database folder (default "mainnetdb")
      --database.dirty string                             set the dirty flag of the database
      --database.inMemory                                 whether the database is only kept in memory and not persisted
      --faucet.blacklistCapacity int                      holds the maximum amount the address blacklist holds (default 10000)
      --faucet.maxTransactionBookedAwaitTimeSeconds int   the max amount of time for a funding transaction to become booked in the value layer (default 5)
      --faucet.powDifficulty int                          defines the PoW difficulty for faucet payloads (default 22)
      --faucet.preparedOutputsCounts int                  number of outputs the faucet prepares (default 126)
      --faucet.seed string                                the base58 encoded seed of the faucet, must be defined if this faucet is enabled
      --faucet.startIndex int                             address index to start faucet with
      --faucet.tokensPerRequest int                       the amount of tokens the faucet should send for each request (default 1000000)
      --fpc.awaitOffset int                               The max amount of time (in seconds) to wait for the next dRNG round after the excected time has elapsed (default 3)
      --fpc.bindAddress string                            the bind address on which the FPC vote server binds to (default "0.0.0.0:10895")
      --fpc.defaultRandomness float                       The default randomness used by FPC when no random is received from the dRNG (default 0.5)
      --fpc.drngInstanceID uint32                         The instanceID of the dRNG to be used with FPC (default 1339)
      --fpc.listen                                        if the FPC service should listen (default true)
      --fpc.querySampleSize int                           Size of the voting quorum (k) (default 21)
      --fpc.roundInterval int                             FPC round interval [s] (default 10)
      --fpc.totalRoundsFinalization int                   The number of rounds opinion needs to stay the same to become final (l) (default 10)
      --gossip.ageThreshold duration                      message age threshold for gossip (default 1m0s)
      --gossip.port int                                   tcp port for gossip connection (default 14666)
      --gossip.tipsBroadcaster.interval duration          the interval in which the oldest known tip is re-broadcast (default 10s)
  -h, --help                                              help for test
      --logger.remotelog.serverAddress string             RemoteLog server address (default "ressims.iota.cafe:5213")
      --mana.allowedAccessFilterEnabled                   list of nodes that consensus mana is allowed to be pledge to
      --mana.allowedAccessPledge strings                  list of nodes that access mana is allowed to be pledged to
      --mana.allowedConsensusFilterEnabled                if filtering on consensus mana pledge nodes is enabled
      --mana.allowedConsensusPledge strings               list of nodes that consensus mana is allowed to be pledge to
      --mana.debuggingEnabled                             if mana plugin responds to queries while not in sync
      --mana.decay float                                  decay coefficient used for Base Mana 2 calculation (default 3.209e-05)
      --mana.emaCoefficient1 float                        coefficient used for Effective Base Mana 1 (moving average) calculation (default 3.209e-05)
      --mana.emaCoefficient2 float                        coefficient used for Effective Base Mana 1 (moving average) calculation (default 0.0057762265)
      --mana.enableResearchVectors                        enable mana research vectors
      --mana.pruneConsensusEventLogsInterval duration     interval to check and prune consensus event storage (default 5m0s)
      --mana.snapshotResetTime                            when loading snapshot reset to current time when true
      --mana.vectorsCleanupInterval duration              interval to cleanup empty mana nodes from the mana vectors (default 30m0s)
      --messageLayer.fcob.averageNetworkDelay int         the avg. network delay to use for FCoB rules (default 2)
      --messageLayer.snapshot.file string                 the path to the snapshot file (default "./snapshot.bin")
      --messageLayer.snapshot.genesisNode string          the node (base58 public key) that is allowed to attach to the genesis message (default "Gm7W191NDnqyF7KJycZqK7V6ENLwqxTwoKQN4SmpkB24")
      --messageLayer.startSynced                          start as synced
      --messageLayer.tangleTimeWindow duration            the time window in which the node considers itself as synced according to TangleTime (default 1m0s)
      --messageLayer.tangleWidth int                      the width of the Tangle
      --network.bindAddress string                        bind address for global services such as autopeering and gossip (default "0.0.0.0")
      --network.externalAddress string                    external IP address under which the node is reachable; or 'auto' to determine it automatically (default "auto")
      --node.disablePlugins strings                       a list of plugins that shall be disabled
      --node.enablePlugins strings                        a list of plugins that shall be enabled
      --rateSetter.initial float                          the initial rate of rate setting (default 20000)
      --scheduler.maxBufferSize int                       maximum buffer size (in bytes) (default 100000000)
      --scheduler.rate string                             message scheduling interval [time duration string] (default "5ms")
      --skip-config                                       Skip config file availability check
      --statement.cleanInterval int                       the time in minutes after which the node cleans the statement registry (default 5)
      --statement.deleteAfter int                         the time in minutes after which older statements are deleted from the registry (default 5)
      --statement.readManaThreshold float                 Value describing the percentage of top mana nodes to accept a statement from (default 1)
      --statement.waitForStatement int                    the time in seconds for which the node wait for receiving the new statement (default 5)
      --statement.writeManaThreshold float                Value describing the percentage of top mana nodes that can write a statement (default 0.7)
      --statement.writeStatement                          if the node should make statements (default true)
  -v, --version                                           version for test
      --webapi.basic_auth.enabled                         whether to enable HTTP basic auth
      --webapi.basic_auth.password string                 HTTP basic auth password (default "goshimmer")
      --webapi.basic_auth.username string                 HTTP basic auth username (default "goshimmer")
      --webapi.bindAddress string                         the bind address for the web API (default "127.0.0.1:8080")
```

**Comment by dessaya** [comment]
Reopening because it is happening again, with this import sequence

```
github.com/iotaledger/goshimmer/client
 imports github.com/iotaledger/goshimmer/plugins/webapi/tools
  imports github.com/iotaledger/goshimmer/plugins/webapi
   imports github.com/iotaledger/goshimmer/plugins/config
```

**Comment by jonastheis** [comment]
This is fixed with #1421 right?
